django/thunderstore/repository/package_upload.py

```diff
@@ -59,19 +60,36 @@ def __init__(self, user, identity, community, *args, **kwargs):
         self.readme: Optional[str] = None
         self.file_size: Optional[int] = None
 
-    def validate_manifest(self, manifest_str):
+    def validate_manifest(self, manifest_str) -> None:
         try:
             manifest_data = json.loads(manifest_str)
         except json.decoder.JSONDecodeError as exc:
             raise ValidationError(f"Unable to parse manifest.json: {exc}")
 
         serializer = ManifestV1Serializer(
             user=self.user,
-            uploader=self.identity,
             data=manifest_data,
         )
         if serializer.is_valid():
+            # Identity validation
+            data = serializer.validated_data
+            uploader = UploaderIdentity.get_or_create_for_user(
+                data["author_name"], self.user
+            )
+            if not uploader.can_user_upload(self.user):
+                raise ValidationError(
+                    f"Missing privileges to upload under author {self.uploader.name}"
+                )
+            reference = PackageReference(
+                data["author_name"], data["name"], data["version_number"]
```

Comment:
I'd use the `UploaderIdentity.name` for the first part here just to avoid any weird bugs in the future by botched refactoring.